src/Kopernicus/RuntimeUtility/RuntimeUtility.cs

```diff
@@ -276,6 +276,13 @@ void Start()
                         Destroy(city.gameObject);
                     }
                 }
+                var stockLaunchSitesField = PSystemSetup.Instance.GetType().GetField("stocklaunchsites", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
Instead of using the name of the field, you should detect it based on it's position and fieldtype, since the names of private fields can be subject to obfuscation (at least in the past this happened on some builds). Something like `GetFields(BindingFlags.NonPublic | BindingFlags.Instance).FirstOrDefault(f => f.FieldType == typeof(LaunchSite[]))` should work even if the name of the field should change.

Also, I would prefer if you could use `typeof(PSystemSetup)` instead of `PSystemSetup.Instance-GetType()`. Saves some calls and looks nicer IMO.

Reply:
`typeof(PSystemSetup)` is certainly superior in this scenario, `.GetType` is just used more often because of how reflection is traditionally employed. I'll correct this, thank you!

I understand your point with name of the field but I disagree with the conclusion. `Assembly-Csharp.dll` already does get put through an obfuscator, btw  (the cheap one that puts no-op switch statements all over the place), which doesn't mean much since they could switch some time or consider it a done deal. The backing field's name is substantially more durable than it being the only private `LaunchSite[]` field in the class, though. `.FirstOrDefault(f => f.FieldType == typeof(LaunchSite[]))` would silently introduce a regression (or worse) if the developers added another private field of the same type.

If Kopernicus weren't version locked, it might be necessary to add exception handling, though. As is, it will be easier to resolve if it throws an exception while preparing for a future version. All in all, if that field changes, a dev will have to look into it, probably by inspecting the source.

Reply:
Ok, then let's see what will happen.

However, could you please add a `try { } catch { }` around the code? Just to prevent it from causing the whole mod to fail.